src/daft-plan/src/logical_ops/sink.rs

```diff
@@ -56,11 +59,17 @@ impl Sink {
                 fields
             }
             #[cfg(feature = "python")]
-            SinkInfo::CatalogInfo(..) => {
-                vec![
-                    // We have to return datafile since PyIceberg Table is not picklable yet
-                    Field::new("data_file", DataType::Python),
-                ]
+            SinkInfo::CatalogInfo(catalog_info) => {
+                match catalog_info.catalog {
+                    CatalogType::Iceberg(_) => {
+                        vec![
+                            // We have to return datafile since PyIceberg Table is not picklable yet
+                            Field::new("data_file", DataType::Python),
+                        ]
+                    }
+                    CatalogType::DeltaLake(_) => vec![Field::new("data_file", DataType::Python)],
+                    CatalogType::Lance(_) => vec![Field::new("fragments", DataType::Python)],
+                }
```

Comment:
Why is this change necessary?

Reply:
Lance actually doesn't return a `data_file` field. This is just a drive-by fix of that